python_modules/libraries/dagster-dbt/dagster_dbt/asset_utils.py

```diff
@@ -956,6 +972,18 @@ def get_asset_check_key_for_test(
     )
 
 
+def get_checks_on_sources_upstream_of_selected_assets(
+    assets_def: AssetsDefinition, selected_asset_keys: AbstractSet[AssetKey]
+) -> AbstractSet[AssetCheckKey]:
+    sources_upstream_of_selected_assets = {
+        dep.asset_key for key in selected_asset_keys for dep in assets_def.specs_by_key[key].deps
+    } - assets_def.specs_by_key.keys()
```

Comment:
nit: I think this can just be `assets_def.keys`

Reply:
I don't think so in this case; since we filter down to only the selected asset keys in the computation from the keys parameter. specs_by_key explicitly goes to the "global set" on the assets def.